[gh-pages] [ZEPPELIN-2597] adding technologies and deployments (Main Page)

soralee opened the pull request:
### What is this PR for?
This PR is for Improving more comfortable guidance for point of user view in the main zeppelin homepage.
To sum up, Technologies section is consist of 3 interpreters most people search or want (already talking about this on the mailing list.) and, Deployments section describes single or multiple user setting.


### What type of PR is it?
[Improvement]


### What is the Jira issue?
* [ZEPPELIN-2597](https://issues.apache.org/jira/browse/ZEPPELIN-2597)


### How should this be tested?
1. run `bundle exec jekyll serve --watch` in this branch
2. open `localhost:4000`


### Screenshots (if appropriate)
- Look `Technologies` and `Deployments` sections
![peek 2017-06-05 15-14](https://cloud.githubusercontent.com/assets/8110458/26772663/d5ba7cbe-4a01-11e7-8715-03569d00509a.gif)


### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes, this is.

soralee commented:
@AhyoungRyu @1ambda @Leemoonsoo, please review again!

sravan-s commented:
I think it would be better if the font sizes inside top box(technologies) and bottom(deployments) were of same size and had a uniform margin / padding scheme. Also about **use now** and **learn more** buttons

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+          </label>
+          <label class="content-text">
+            Zeppelin supports Multi-user Support w/ LDAP. Let's configure Zeppelin for your yarn cluster.<br/>
+            <a href="./docs/0.8.0-SNAPSHOT/usage/other_features/personalized_mode.html" class="user-button" style="margin-top:15px">LEARN MORE</a>
```
1. Need more `margin-top`.  25px would be proper. 
2. would be nice to add `padding-bottom: 15px` for `.advertise .panel-content-user` (decreasing 10px) since we add `10px` to margin-top.



![image](https://cloud.githubusercontent.com/assets/4968473/26774914/7a9daed0-4a0d-11e7-972a-be1843eab5e5.png)

soralee replied:
I pushed one commit, could you confirm this?

1ambda replied:
DONE.

soralee replied:
Thanks!

1ambda commented on assets/themes/zeppelin/css/style.css:
```diff
@@ -1134,7 +1195,225 @@ and (max-width: 1024px) {
+}
+
+.advertise h2 {
+  font-family: "Helvetica";
```
2. Use thin font for `TECHNOLOGIES` and `DEPLOYMENTS`. 

- It would be nice to have less font-weight for not important texts so that users can focus on actual contents. not the title.

![image](https://cloud.githubusercontent.com/assets/4968473/26775268/1f6cfe4c-4a0f-11e7-9474-fb32167a4c8c.png)

![image](https://cloud.githubusercontent.com/assets/4968473/26775273/256ef598-4a0f-11e7-9e00-9aec76efdde3.png)

soralee replied:
I pushed second commit, could you confirm this?

1ambda replied:
Checked.

1ambda commented on assets/themes/zeppelin/css/style.css:
```diff
@@ -1134,7 +1195,225 @@ and (max-width: 1024px) {
+
+.advertise {
+  background: #FFF;
+  padding-bottom: 15px;
```
3. more `padding-bottom` for `.advertise`. It's too small compared to `padding-top`. 

`padding-bottom: 35px;` would be proper.

#### Before

![image](https://cloud.githubusercontent.com/assets/4968473/26775363/995a626c-4a0f-11e7-9086-898cf766cc18.png)

#### After

![image](https://cloud.githubusercontent.com/assets/4968473/26775387/af338ba4-4a0f-11e7-948e-81b95ff8f480.png)

soralee replied:
I pushed one commit, could you confirm this?

1ambda replied:
checked.

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+-->
+{% include JB/setup %}
+<div class="advertise">
+  <div class="advertise container">
```
should use a different class name instead of `advertise`. It's duplicated with the line just above.

soralee replied:
I pushed fourth and fifth commit. I've changed class name and filename as well.
this changed commit name is `change advertise.md -> technologies_deployments.md` and `remove file`.

1ambda replied:
still not resolved. I added new comment

soralee replied:
ah, sorry! I totally misunderstood that!
Let me rename class name of this line (`<div class="advertise container">`).

soralee replied:
I changed `div class` name, which is `section container`.

soralee replied:
ah, I haven't find your comment, but there is in the 11 issue. 
Let's improve again.

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+      <div class="border col-md-6 col-sm-6">
+        <p class="panel-content-user">
+          <label style="width: 100%;">
+            <span class="glyphicon glyphicon-user" style="font-size:30px;position:relative;width:100%;text-align:center;"></span>
```
MINOR: Can we use http://fontawesome.io/icon/user/ instead of `glyphicon-user`? 

I know they have the same look. But it would be nice to use the same font classes with `fa-users` for consistency.

soralee replied:
I pushed second commit, could you confirm this?

1ambda replied:
checked.

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+      </div>
+      <div class="border col-md-4 col-sm-4" style="height:200px;">
+        <div class="panel-content">
+          <label class="_default-text" style="font-size:56px;">SQL</label>
```
4. not sure how we can improve this `SQL` `panel-content`.  But here are few suggestions. 

- Use a different `font-family` and `font-weight`
- `Roboto` with `font-weight: 100` is the best for me.
- you can test more fonts in https://fonts.google.com/?category=Sans+Serif,Monospace&thickness=2

### `SQL` text with `font-weight: 100`

![image](https://cloud.githubusercontent.com/assets/4968473/26776777/9924f428-4a15-11e7-86d8-87ed90970d22.png)

### `SQL` text with `font-weight: 500` (normal)

![image](https://cloud.githubusercontent.com/assets/4968473/26776740/67033d4c-4a15-11e7-9676-3f7305682394.png)

1ambda replied:
ping @soralee

soralee replied:
![image](https://cloud.githubusercontent.com/assets/8110458/26820322/6c94b452-4add-11e7-81d0-1696e60d185c.png)
@1ambda How about this font?
SQL with `font-family: 'Raleway', sans-serif; weight:300;`

1ambda replied:
Looks nice. Could you attach a screenshot for `weight: 300`?

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+      </div>
+      <div class="border col-md-4 col-sm-4" style="height:200px;">
+        <div class="panel-content">
+          <label class="_default-text" style="margin-top:65px;">
```
5. need to increase `margin-top` for the python logo. It's not aligned with the other image (spark) and text (SQL)

### Current

![image](https://cloud.githubusercontent.com/assets/4968473/26777034/aa6941fc-4a16-11e7-8c3f-e88638d2d2ce.png)




### Fixed: `margin-top: 75px`

![image](https://cloud.githubusercontent.com/assets/4968473/26777023/9d70f422-4a16-11e7-9cc8-49038131da7d.png)

soralee replied:
I pushed one commit, could you confirm this?

1ambda replied:
checked.

1ambda commented on advertise.md:
```diff
@@ -0,0 +1,86 @@
+          <label class="_default-text" style="margin-top: 20px;">
+            <img src="./assets/themes/zeppelin/img/spark_logo.jpg" width="140px">
+          </label>
+          <label class="_hover-text">
```
6. increase padding for `_hover_text`

`padding: 13px` might be proper.

### Current

![image](https://cloud.githubusercontent.com/assets/4968473/26777263/90a3581a-4a17-11e7-9c9e-b3a5708569e6.png)


### Fixed

![image](https://cloud.githubusercontent.com/assets/4968473/26777254/86f3c8d6-4a17-11e7-8dfc-d4e7a3566219.png)

sravan-s replied:
@1ambda @soralee , I think, **USE NOW** is a bit bolder than she should be. What do you guys think?

1ambda replied:
@sravan-s Could you attach an example? 

### Developemnt Environment

1. checkout
2. `cd docs`
3. `bundle install` (make sure that you have ruby 2.1.0+ and bundle)
4. `bundle exec jekyll serve --watch`
5. open `localhost:4000`

soralee replied:
@sravan-s, I think this reason is difference of OS. My OS is Ubuntu. 
Anyway, those font of buttons look too big.

1ambda replied:
checked.

1ambda commented:
7. Improve layout for mobile browsers. 

7-1. each `panel-content` in the **TECHNOLOGIES** section doesn't have margin-top or bottom. It's not consistent compared to `panel-content-user` in  the **DEPLOYMENTS** section.

![image](https://cloud.githubusercontent.com/assets/4968473/26777399/036b2864-4a18-11e7-95a1-f6bc2b976236.png)

7-2. It would be nice to add `padding-left` and `padding-left` to `.advertise` (30px might be proper.)

### Current

![image](https://cloud.githubusercontent.com/assets/4968473/26777452/44b842ac-4a18-11e7-8439-5af3cde788b8.png)

### After

![image](https://cloud.githubusercontent.com/assets/4968473/26777440/346535fe-4a18-11e7-9bd6-780b15fbffbe.png)

soralee commented:
@sravan-s, I've fixed to match font size as your mention. Regarding of `uniform margin / padding scheme` I tried to fit the box ratio. could you check this if you have time?

soralee commented:
@1ambda , I've update regarding of  `7.Improve layout for mobile browsers.` could you confirm, again?

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+-->
+{% include JB/setup %}
+<div class="tech_deploy">
+  <div class="tech_deploy container">
```
@soralee What i mentioned is, using different name in different `DIV`s. for example

```HTML
<div class=tech-deploy-outbox>
  <div class=tech-deploy-innerbox>
```

not using the same class name `tech_deploy`.

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+          <label class="_default-text" style="margin-top: 20px;">
+            <img src="./assets/themes/zeppelin/img/spark_logo.jpg" width="140px">
+          </label>
+          <label class="_hover-text">
```
8. Can we apply `min-height` for `_hover-text` class? so that a change in text can't affect the layout of the **USE NOW** button.

soralee replied:
Sure, I just added!

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+            <img src="./assets/themes/zeppelin/img/spark_logo.jpg" width="140px">
+          </label>
+          <label class="_hover-text">
+            Spark interpreter group which consists of five interpreters which are Spark, Pyspark, SparkR, Spark SQL, Dependency loader.<br/>
```
9-1 IMO, we don't need to describe or use Zeppelin specific terms on the front page for beginners.

> Zeppelin supports Spark, PySpark, SparkR, SparkSQL with dependency loader. 

would be proper.

soralee replied:
changed. please check again (:

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+            <img src="./assets/themes/zeppelin/img/python-logo-generic.svg" width="200px">
+          </label>
+          <label class="_hover-text">
+            Python is supported with Zeppelin. Also integrated Matplotlib with Python & Pyspark interpreter, and Conda is now available in Zeppelin.<br/>
```
9-2 Let's simplify this sentence

> Python is supported with Matplotlib, Conda, PandasSQL and PySpark integrations.

soralee replied:
changed as well. please check again (:

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+        <div class="panel-content">
+          <label class="_default-text" style="font-size:56px;">SQL</label>
+          <label class="_hover-text">
+            JDBC interpreter lets you create a JDBC connection to any data sources seamlessly. Postgresql, Mysql, MariaDB, Redshift, Apache Hive and so on.<br/>
```
9.3 Let's change this text little bit.

> Zeppelin lets you connect any JDBC data sources seamlessly. Postgresql, Mysql, MariaDB, Redshift, Apache Hive and so on.

soralee replied:
changed as well. please check again (:

1ambda commented on technologies_deployments.md:
```diff
@@ -0,0 +1,86 @@
+    </div>
+    <div class="col-md-12 col-sm-12 col-xs-12 text-center">
+      <p style="text-align:center; margin-top: 20px; font-size: 17px; color: gray; font-weight: 200; font-style: italic; padding-bottom: 0;">
+        See more details in Zeppelin supports 20+ different interpreters.
```
10: (MINOR). can we move these kind of big inline styles into CSS? I don't mean moving all inline styles into CSS. just big inline styles like **LINE: 52** so that others can change style easily instead of modifying unfamiliar a md file.

1ambda commented:
11. Adjustment margin among components

- `padding-bottom: 30px` for `.tech_deploy h2`
- remove the css `tech_deploy` class in `technologies_deployments.md` LINE16: `tech_deploy container` -> `container`
- `padding-top: 22px` and `padding-bottom: 40px` for `.tech_deploy`
- remove `padding-bottom` and `margin-top` in `technologies_deployments.md: LINE 52`
- remove `padding-bottom` and `padding-top` in the `.tech_deploy p` class. (DON'T SET to 0, just remove the style)
- `margin-top: 25px`, `margin-bottom: 30px` for `.tech_deploy p`
- inline style `padding-bottom: 8px;` for **DEPLOYMENTS** `h2`

### Current

![image](https://cloud.githubusercontent.com/assets/4968473/26810862/016a8956-4aa9-11e7-8f8a-d8e6a1d13f7c.png)

### After

![image](https://cloud.githubusercontent.com/assets/4968473/26810957/bd8cb5f0-4aa9-11e7-9cda-79d2b99ac127.png)

1ambda commented:
Thanks for your great effort @soralee. LGTM! except 

* [ ] - comments 4
* [ ] - comments 8
* [ ] - comments 9-1, 9-2, 9-3
* [ ] - comments 10
* [ ] - comments 11

soralee commented:
I really thank @1ambda and  your a kind explanation as well!
I'll update the following list and let you know that. 
```
 - comment 4 (SQL text font styling)
 - comment 8
 - comments 9-1, 9-2, 9-3
 - comment 10
 - comment 11
```

soralee commented:
@1ambda, I just finished the list about all your comment.
For help to easy your review, I involved comment numbers in title message when I commit.
I appreciate for taking care of this PR, all detailed comment as well!

soralee commented:
some improving list
- arranged to same line each 'use now' button in the Technologies.
- changed contents font style in Technologies and Deployments. 
- removed bold font of buttons in the Deployments.

1ambda commented:
LGTM. 

## Chrome on OSX

![image](https://cloud.githubusercontent.com/assets/4968473/26826728/0f6df730-4af5-11e7-914f-9154b5e8ca87.png)

## Safari on OSX

![image](https://cloud.githubusercontent.com/assets/4968473/26826748/20c1f842-4af5-11e7-8cac-d187b3a8102a.png)

## Old version Firefox (v31) on OSX

![image](https://cloud.githubusercontent.com/assets/4968473/26826771/33cc509a-4af5-11e7-9bfa-f9fd4867b48b.png)

soralee commented:
@1ambda Thanks a lots of! you're the best! :+1:

AhyoungRyu commented:
Thanks for the great work @soralee! And also thanks for detailed review @1ambda. 
Tested and looks nice! 
I really like the most of parts. Some minor suggestion is

#### Technologies
 - how about using same border color with "Deployment" box `#999` instead of `#4881B3`?
<img width="997" alt="screen shot 2017-06-06 at 7 36 05 pm" src="https://user-images.githubusercontent.com/10060731/26859880-7f3df818-4aef-11e7-9364-a0274c2642bf.png">

#### Deployment
 -  how about putting "LEARN MORE" buttons in each deployment section box in the middle instead of right side? 
 - and also we can remove the single & multi user font awesome icons since the text ("single" and "multi") itself already explain what "single-user" and "multi-user" are meaning for.

How about this? It's just my opinion. Please have a look and feel free to share your thought :)

<img width="990" alt="screen shot 2017-06-06 at 7 25 45 pm" src="https://user-images.githubusercontent.com/10060731/26859800-0d9d5cf8-4aef-11e7-804c-3f93a1ae035d.png">

soralee commented:
@AhyoungRyu, Thanks for review and suggestions!
Sure, I can change the border color in Technologies section and, I can remove icons in Deployments section.

IMO, about Deployments section, actually the current structure is good to me (including icons). 
And especially the buttons, it would be better to keep the current structure to match the buttons in the above technologies.

![screenshot from 2017-06-07 12-03-09](https://user-images.githubusercontent.com/8110458/26860704-5625eab8-4b7a-11e7-8f04-a9b509b8ab8c.png)

How about this? :slightly_smiling_face:

soralee commented:
or to decorate in Deployments section, How about this structure?

![image](https://user-images.githubusercontent.com/8110458/26861081-70ce8f80-4b7c-11e7-881d-c7fa9e6d95c5.png)

AhyoungRyu commented:
Yeah second option is better for me :)

soralee commented:
@AhyoungRyu, Thanks for accepting :+1: 
Let me improve this and let you know.

sravan-s commented:
@soralee , sorry, I am having some trouble installing jekyll and stuff in my machine, but your screenshots looks good :)

soralee commented:
@AhyoungRyu I updated as your mentions.
![image](https://user-images.githubusercontent.com/8110458/26912241-117e2a68-4c4e-11e7-9827-6ee01e514ad3.png)

soralee commented:
Ping, ready to review!

1ambda commented:
@soralee Since https://github.com/apache/zeppelin/pull/2371 was merged, we can fix the URL names in this PR too.

- for spark button: `${PROPER_PREFIX_FOR_0.8.0}/quickstart/spark_with_zeppelin.html`
- for sql button: `${PROPER_PREFIX_FOR_0.8.0}/quickstart/sql_with_zeppelin.html`
- for python button: `${PROPER_PREFIX_FOR_0.8.0}/quickstart/python_with_zeppelin.html`
- single user button: `${PROPER_PREFIX_FOR_0.8.0}/quickstart/install.html`
- multi user button: `${PROPER_PREFIX_FOR0.8.0}/setup/basics/multi_user_support.html`

soralee commented:
@1ambda Thanks for arranging list
Updated it.

1ambda commented on technologies_deployments.md:
```diff
@@ -73,7 +73,7 @@ limitations under the License.
           </label>
           <label class="content-text">
             Local Spark, 6 Built-in visualizations, Display system, Dynamic form, Multiple backends are supported.<br/>
-            <a href="./docs/0.8.0-SNAPSHOT/quickstart/tutorial.html" class="user-button">LEARN MORE</a>
```
Is is ok to remove `.`?

soralee replied:
yeah, don't worry. It works well!

soralee commented:
@1ambda, How about that connecting the latest version documentation? we'd better to guide the latest version to user.

```
spark - http://zeppelin.apache.org/docs/0.7.2/interpreter/spark.html
sql - http://zeppelin.apache.org/docs/0.7.2/interpreter/jdbc.html
python - http://zeppelin.apache.org/docs/0.7.2/interpreter/python.html

single user - http://zeppelin.apache.org/docs/0.7.2/install/install.html
multi user - http://zeppelin.apache.org/docs/0.7.2/security/shiroauthentication.html
```

1ambda commented:
Looks good to me and agree that use old links unile we deploy 0.8.0

soralee commented:
Yeah, I updated it (: 
thanks @1ambda

1ambda commented:
@soralee Could you resolve conflict?

soralee commented:
@1ambda Fixed!

1ambda commented:
Looks good to me. Thanks for your contribution. 

Merge if no more discussion.

![image](https://user-images.githubusercontent.com/4968473/27670824-6324fc98-5cca-11e7-9e57-c325663472dc.png)
